tests/test_benchmark/conftest.py

```diff
@@ -0,0 +1,12 @@
+from __future__ import annotations
+
+import pytest
+
+import mteb
+
+from .task_grid import MOCK_TASK_REGISTRY
+
+
+@pytest.fixture
+def mock_mteb_get_task(monkeypatch):
+    monkeypatch.setattr(mteb.overview, "TASKS_REGISTRY", MOCK_TASK_REGISTRY)
```

Comment:
I don't think we actually want this - we should be able to test mteb.get_tasks() works as intended with the real tasks.

Reply:
But I see that you must have added this to get mteb.get_task() to work within the model_encode - in which case I am probably more a fan of the previous approach (sorry about that), since this change is likely to cause tests which behaves unexpectedly

Reply:
I understand your concerns, but I believe this approach is better because it simplifies the code significantly and allows us to gather more information about tasks. In the future, metadata (e.g., instructions) could be used more extensively, and using `mteb.get_task` inside `model_encode` would be a much simpler solution than passing parameters throughout the code.

Reply:
Hmm I see the point. I also really like the code simplification. 

I think a solution is to change the interface such that it only takes in promptType and task_name (i.e. remove task_type). Then we can introduce a SentenceTransformerWrapper (similar to what is done in #1225), which can then do all the heavy lifting. For the test, we then simply create a wrapper that uses the MOCK_REGISTRY instead of the mteb.get_tasks().

This keeps the simplicity, moved sentence transformer specific transformations to a separate wrapper and removes the need for monkeypatching in tests

let me know what you think about this approach

Reply:
You suggest moving `model_encode` to separate wrapper?

Reply:
yes, right now it is really intended for SentenceTransformers (cc @isaac-chung just to get a second opinion)

Reply:
Hmm, I might be missing something but I'm not following how this significantly simplifies the code?

Reply:
Using `mteb.get_task` in `model_encode` or wrappers is simpler than passing arguments from tasks or evaluators.

Reply:
Not sure what part you mean, but using `mteb.get_task` vs passing task_type down through the call stack does notably reduce the information handling (and avoids accidental defaults overwriting e.g. task_type)

Reply:
Ah I see, thanks!